src/Symfony/Component/Cache/Traits/ApcuTrait.php

```diff
@@ -71,9 +71,24 @@ protected function doHave($id)
      */
     protected function doClear($namespace)
     {
-        return isset($namespace[0]) && class_exists('APCuIterator', false) && ('cli' !== PHP_SAPI || ini_get('apc.enable_cli'))
-            ? apcu_delete(new \APCuIterator(sprintf('/^%s/', preg_quote($namespace, '/')), APC_ITER_KEY))
-            : apcu_clear_cache();
+        if (isset($namespace[0]) && ('cli' !== PHP_SAPI || ini_get('apc.enable_cli'))) {
+            if (class_exists('APCuIterator', false)) {
+                $keysToDelete = new \APCuIterator(sprintf('/^%s/', preg_quote($namespace, '/')), APC_ITER_KEY);
+            } else {
+                $keysToDelete = array();
+                $cacheInfo = apcu_cache_info();
+                foreach ($cacheInfo['cache_list'] as $cacheEntry) {
+                    $cacheKey = $cacheEntry['info'];
+                    if (strpos($cacheKey, $namespace) === 0) {
+                        $keysToDelete[] = $cacheKey;
+                    }
+                }
+            }
+
+            return apcu_delete($keysToDelete);
+        }
+
+        return apcu_clear_cache();
```

Comment:
would be great if this return could be first, so that we can remove one level of indentation above

Reply:
done